modules/outbound-command-event-handler/src/domain/bulk_transaction_agg.ts

```diff
@@ -0,0 +1,154 @@
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+ * Modusbox
+ - Vijay Kumar Guthi <vijaya.guthi@modusbox.com>
+ --------------
+ ******/
+
+'use strict';
+
+import { ILogger } from '@mojaloop/logging-bc-public-types-lib';
+import { BaseAggregate, IEntityStateRepository } from '@mojaloop/sdk-scheme-adapter-private-shared-lib';
+import { BulkTransactionEntity, BulkTransactionState } from './bulk_transaction_entity';
+import {
+    IndividualTransferEntity,
+    IndividualTransferState,
+} from './individual_transfer_entity';
+import { IBulkTransactionEntityRepo } from '../types';
+
+export class BulkTransactionAgg extends BaseAggregate<BulkTransactionEntity, BulkTransactionState> {
+    // TODO: These counts can be part of bulk transaction entity?
+    // private _partyLookupTotalCount?: number;
+    // private _partyLookupSuccessCount?: number;
+    // private _partyLookupFailedCount?: number;
+    // // NEXT_STORY
+    // private _bulkBatches: Array<BulkBatchState>; // Create bulk batch entity
+    // private _bulkQuotesTotalCount: number;
+    // private _bulkQuotesSuccessCount: number;
+    // private _bulkQuotesFailedCount: number;
+    // private _bulkTransfersTotalCount: number;
+    // private _bulkTransfersSuccessCount: number;
+    // private _bulkTransfersFailedCount: number;
+
+    // constructor(
+    //   bulkTransactionEntity: BulkTransactionEntity,
+    //   entityStateRepo: IEntityStateRepository<BulkTransactionState>,
+    //   logger: ILogger
+    // ) {
+    //     super(bulkTransactionEntity, entityStateRepo, logger);
+    // }
+
+    static async CreateFromRequest(
+        /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+        request: any,
+        entityStateRepo: IEntityStateRepository<BulkTransactionState>,
+        logger: ILogger,
+    ): Promise<BulkTransactionAgg> {
+    // Create root entity
+        const bulkTransactionEntity = BulkTransactionEntity.CreateFromRequest(request);
+        // Create the aggregate
+        const agg = new BulkTransactionAgg(bulkTransactionEntity, entityStateRepo, logger);
+        // Persist in the rep
+        await agg.store();
+        // Create individualTransfer entities
+        if(Array.isArray(request?.individualTransfers)) {
+            // TODO: limit the number of concurrently created promises to avoid nodejs high memory consumption
+            await Promise.all(request.individualTransfers.map((individualTransfer: any) =>
+                agg.addIndividualTransferEntity(IndividualTransferEntity.CreateFromRequest(individualTransfer))));
+        }
+        // Return the aggregate
+        return agg;
+    }
+
+    static async CreateFromRepo(
```

Comment:
Similar to point 4 above, we can apply the same logic here with some re-wording...

4. We should not be creating an Aggregate from a Repo, but rather using an Aggregate to process a request